### PR TITLE
INFR-461: Fix broken start function - too many PIDs

### PR DESCRIPTION
**Pull request by senilegenius**

### CODE REVIEW

File: init.d/passenger-datadog-monitor.sh [hunk 1]
```diff
@@ -36,7 +40,19 @@ start() {
...
+      sleep "${SLEEPTIME}"
+      (( SLEEPCOUNT-- ))
+      if [[ "${SLEEPCOUNT}" -eq 0 ]]; then
+        echo "There are multiple ${PDM} processes running, please check."
+        exit 1
+      fi
+      PID="$(get_pid)"
+      echo "${PID}" > "${PIDFILE}"
+      PIDCOUNT="$(wc -l ${PIDFILE} | awk '{print $1}')"
+    done
```
Comment (jlott) [inline init.d/passenger-datadog-monitor.sh:55]: You could do this more efficiently and without that `get_pid()` function with something like this:

```SLEEPCOUNT="10"
while true; do
  mapfile -t PID < <(pgrep -f -x "${PDMDIR}/${PDM}")
  if [[ "${#PID[@]}" != 1 ]]; then
    if [[ "${SLEEPCOUNT}" == 0 ]]; then
      echo "ERROR: Expected 1 process ID for ${PDM} but found ${#PID[@]}."
      echo "Process ID's:"
      echo "${PID[@]}"
      exit 1
    fi
    sleep .5
    (( SLEEPCOUNT-- ))
  else
    echo "${PID[@]}" > "${PIDFILE}"
    break
  fi
done```

Comment (jlott) [reply init.d/passenger-datadog-monitor.sh:55]: Note that `mapfile` is bash 4.x, must not be in posix mode, and may use temporary files... but I don't think that will be a problem for any of our infrastructure.

File: init.d/passenger-datadog-monitor.sh [hunk 2]
```diff
@@ -36,7 +40,19 @@ start() {
   su - root -c "/usr/bin/nohup ${PDMDIR}/${PDM} > ${LOGFILE} 2>&1 &"
   if [[ "${RETVAL}" -eq 0 ]]; then
     PID="$(get_pid)"
-    echo "${PID}" >"${PIDFILE}"
+    echo "${PID}" > "${PIDFILE}"
```
Comment (akanix42) [inline init.d/passenger-datadog-monitor.sh:43]: @jlott @senilegenius I believe we settled on no spaces between the `>` and the path, right?
For reference, https://github.com/square-root/k8s-config/blob/e301760ed2ba9c35ae90b0352cb13f879c303bce/bootstrap/setup#L108

File: init.d/passenger-datadog-monitor.sh (hunk 1 above)
Comment (akanix42) [reply init.d/passenger-datadog-monitor.sh:55]: Oooh bash 4! 👏

File: init.d/passenger-datadog-monitor.sh (hunk 2 above)
Comment (jlott) [reply init.d/passenger-datadog-monitor.sh:43]: I personally tend to write that kind of thing with spaces just without even thinking about it. Would need to check our style guide to see if it has anything to say there.

Tangent: We might get to the point soon where we should fork Googles entire style guide and start tweaking it for our own uses in these cases (assuming it doesnt have an opinion on this particular thing)

Comment (senilegenius) [reply init.d/passenger-datadog-monitor.sh:43]: Y'all need to decide.

Space or no space after redirect?

Comment (jlott) [reply init.d/passenger-datadog-monitor.sh:43]: Ok I remember our conversations about this now. No space, @senilegenius 👍

File: init.d/passenger-datadog-monitor.sh (hunk 1 above)
Comment (senilegenius) [reply init.d/passenger-datadog-monitor.sh]: @jlott and I discussed this in DMs a bit.
I've made the suggested change to use `mapfile` and an actual array, all of which is outside of the work of the ticket.
@jlott @akanix42 please re-review.

**Pull request merged**